integration/networktest/traffic/runner.go

```diff
@@ -0,0 +1,160 @@
+package traffic
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/obscuronet/go-obscuro/integration/networktest"
+)
+
+// Runner is used to simulate traffic against a durationTrafficTest network
+// When Start() is called then simulated traffic will begin against the provided NetworkConnector, until Stop() is called.
+// Then Verify() can be used to check the simulation worked as expected (it might check receipts, balances, response times etc.)
+type Runner interface {
+	// Start is only expected to be called once for the lifetime of the Runner
+	Start(network networktest.NetworkConnector) error
+	Stop()
+	RunData() RunData
+	Name() string // brief name (used in log file naming etc., avoid spaces)
+}
+
+// RunData builds up data about the simulated actions during the traffic run
+// - this can be validated as part of an automated durationTrafficTest or perhaps statistics or visualisations created etc.
+// Note: this data can be combined with the runner's data and the network to verify post-state
+type RunData interface {
+	ActionEvents() []ActionRecord // get a list of completed actions
+	RunDescription() string       // few words describing the simulation that was run
+}
+
+type Action interface {
+	Name() string
+}
+
+const (
+	SendTransaction = "SendTransaction"
+)
+
+type ActionRecord interface {
```

Comment:
I'm a big fan of this abstraction.
I'm wondering if we could take this abstraction one step further, and get rid of some complexity.

For example the: `prepareSim` in the `randomTransactingSimRunner`.

A test itself could be a sequence of Actions performed in series of parallel.

- `UserActionItem` could contain the action logic to execute whatever it has to execute, and also to verify that it was executed successfully.

The "deploy" step in the prepareSim would be a normal Action.

-`OpsActionItem` - could be that restarting of the enclave

These actions could be wired to run in series or parallel. 
If in series, then the result of the previous execution should be available to the next